packages/gateway/src/features/user/root-resolvers.ts

```diff
@@ -309,6 +310,44 @@ export const resolvers: GQLResolver = {
       }
       return true
     },
+    async changePhone(
+      _,
+      { userId, phoneNumber, nonce, verifyCode },
+      authHeader
+    ) {
+      if (!isTokenOwner(authHeader, userId)) {
+        return await Promise.reject(
+          new Error(
+            `Change phone is not allowed. ${userId} is not the owner of the token`
+          )
+        )
+      }
+      try {
+        await checkVerificationCode(nonce, verifyCode)
+      } catch (err) {
+        logger.error(err)
+        return await Promise.reject(
+          new Error(`Change phone is not allowed. Error: ${err}`)
+        )
```

Comment:
```suggestion
        throw new Error(`Change phone is not allowed. Error: ${err}`)
```

You can just throw an error to get the same result as we're inside an async context

Reply:
There were other functions in root-resolvers like this, so I leave it as it is right now.

Reply:
Fair enough! I added a tech debt task regarding both of these comments: https://github.com/opencrvs/opencrvs-core/issues/2520